### PR TITLE
YD-211 Completed localization support

### PR DESCRIPTION
**Pull request by Bert-R**
* All localizable resources are now available in Dutch
* Implemented language fallback
* Implemented capability to fetch localized resources for the apps
* Now returns the Content-Language header, to show the result of language negotiation

### CODE REVIEW

File: core/src/main/resources/templates/email/buddy-invitation-body_nl.vm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<html>
```
Comment (aristotelos) [inline core/src/main/resources/templates/email/buddy-invitation-body_nl.vm]: Maybe add lang="nl"? (See https://www.w3.org/International/questions/qa-http-and-lang, some browsers (in 2011?) did not take the header for processing the content.)

File: core/src/main/resources/messages/messages_nl.properties [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+error.mobile.number.confirmation.code.mismatch=De bevestigingscode ''{1}'' is onjuist voor nummer ''{0}''
+error.mobile.number.confirmation.code.not.set=De bevestigingscode voor nummer ''{0}'' is niet gezet
+error.mobile.number.already.confirmed=Mobiel nummer ''{0}'' is al bevestigd
+error.mobile.number.not.confirmed=Mobiel nummer ''{0}'' is nog niet bevestigd
+
+error.user.overwrite.confirmation.code.mismatch=De bevestigingscode ({1}) voor het overschrijven van het account van nummer ``{0}`` is onjuist
+error.user.overwrite.confirmation.code.not.set=De bevestigingscode voor het overschrijven van het account van nummer ``{0}`` is niet gezet
+
+error.email.sending.failed=Versturen E-mail mislukt: {0}
```
Comment (aristotelos) [inline core/src/main/resources/messages/messages_nl.properties]: E-mail zonder hoofdletter

File: core/src/main/resources/messages/messages_nl.properties [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+error.message.action.not.supported=Actie ''{0}'' wordt niet ondersteund
+error.message.anonymized.user.id.must.be.set=Het VPN login ID mag niet leeg zijn
+
+error.loading.activitycategories.from.file=Fout bij het laden van de activiteitencategorien van disk
+error.activitycategory.not.found=Activiteitencategorie met ID ''{0}'' niet gevonden
+error.activitycategory.not.found.by.name=Activiteitencategorie met de naam ''{0}'' niet gevonden
+
+error.ldapuser.action.unknown=Actie ''{0}'' is onbekend
+error.ldapuser.empty.vpn.login.id=Het VPN login ID mag niet leeg zijn
+error.ldapuser.empty.vpn.password=Het VPN wachtwoord mag niet leeg zijn
```
Comment (aristotelos) [inline core/src/main/resources/messages/messages_nl.properties]: Officieel denk ik VPN-wachtwoord

File: core/src/main/resources/messages/messages_nl.properties [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+error.missing.buddy.id=Het buddy ID ontbreekt in het request
+
+error.cannot.delete.unprocessed.message=Bericht kan niet worden verwijderd omdat het nog niet is verwerkt
+
+error.mobile.number.confirmation.code.mismatch=De bevestigingscode ''{1}'' is onjuist voor nummer ''{0}''
+error.mobile.number.confirmation.code.not.set=De bevestigingscode voor nummer ''{0}'' is niet gezet
+error.mobile.number.already.confirmed=Mobiel nummer ''{0}'' is al bevestigd
+error.mobile.number.not.confirmed=Mobiel nummer ''{0}'' is nog niet bevestigd
+
+error.user.overwrite.confirmation.code.mismatch=De bevestigingscode ({1}) voor het overschrijven van het account van nummer ``{0}`` is onjuist
```
Comment (aristotelos) [inline core/src/main/resources/messages/messages_nl.properties]: Hier en in volgende regel worden andere quotes gebruikt

File: core/src/main/resources/templates/email/buddy-invitation-body_nl.vm [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    <body>
+		<h3>Hi ${buddyName}, Word vriend op Yona!</h3>
+		
+		<p>${message}</p>
+		
+		<p>We hadden het er al even over: ik zie je graag als vriend op Yona. Als je de Yona app nog niet ge�nstalleerd hebt, is dat het begin. Dit zijn de links:</p>
```
Comment (aristotelos) [inline core/src/main/resources/templates/email/buddy-invitation-body_nl.vm]: De i-trema werkt niet. Is dit een fout in Github of in de code?

File: core/src/main/java/nu/yona/server/Translator.java [hunk 6]
```diff
@@ -46,4 +44,24 @@ public String getLocalizedMessage(String messageId, Object[] parameters, Locale
...
+	{
+		return MessageFormat.format(format, determineLocaleInfix());
+	}
+
+	private static Object determineLocaleInfix()
+	{
+		Locale locale = LocaleContextHolder.getLocale();
+		if (locale.equals(Locale.forLanguageTag("en-US")))
+		{
+			return "";
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/Translator.java]: Would it not be more logical to have a subfolder for en-US too?

File: appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
+
+import nu.yona.server.Translator;
+import nu.yona.server.properties.YonaProperties;
+
+@Component
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java]: Maybe add a comment describing that the class does.

File: appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+public class LocalizationFilter implements Filter
+{
+	public static class LocalizationRequestWrapper extends HttpServletRequestWrapper
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java]: The only purpose of this class is to set the Locale I guess? Maybe add a class comment describing what the class does.

File: appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java (hunk 7 above)
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java]: Can't we use [LocaleChangeInterceptor|https://github.com/spring-projects/spring-framework/blob/master/spring-webmvc/src/main/java/org/springframework/web/servlet/i18n/LocaleChangeInterceptor.java] for this with the `setHttpMethods("Accept-Language")` configuration?

File: core/src/main/resources/messages/messages_nl.properties (hunk 3 above)
Comment (Bert-R) [inline core/src/main/resources/messages/messages_nl.properties]: Dan is het ook VPN-login-ID. Ik laat dat aan Jan over. Die moet toch alle teksten nog door.

File: core/src/main/java/nu/yona/server/Translator.java (hunk 6 above)
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/Translator.java]: It's not the subfolder but the extension of the file name (before the dot). The default files contain US English and don't have such a "suffix". E.g. ``message.properties``, while the Dutch version is ``message_nl.properties``.

File: appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java (hunk 7 above)
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java]: I'll add that comment.
The ``LocaleChangeInterceptor`` sets the locale based on a query parameter, thus allowing to influence the request through the URL. That's very different from what the ``LocalizationFilter`` does.

File: appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java (hunk 8 above)
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/appresources/LocalizationFilter.java]: It does a bit more, see the added comment.

**Comment by Bert-R** [comment]
Addressed the comments.

**Pull request merged**